app/models/task.rb

```diff
@@ -77,7 +77,9 @@ class Task < ApplicationRecord # rubocop:disable Metrics/ClassLength
     where.missing(:task_contribution)
   }
   scope :pending_contribution, lambda {|user|
-    joins(:task_contribution).where(user:, task_contribution: {status: :pending})
+    # Since accessing any contribution that belongs to a non-public task doesn't work, we filter for the parent visibility.
+    # TODO: Do we want to keep this behavior?
```

Comment:
I'm not sure. There are good arguments for both keeping the current implementation and changing it.

Pros of keeping the current implementation: If the original task's visibility is changed to private accidentally, restoring its visibility would also restore access to the contribution.

Cons: The contributions could reappear after a significant time period, making them less useful.

Reply:
That's true, thanks for your input. As discussed, I created #1633 as a follow-up issue.